### ISSUE TITLE
python dictionaries don't preserve order, support duplicate keys

### ISSUE THREAD

**Issue by tzodyaq** [rossengeorgiev/vdf-parser#3]
Using the following

```python
import vdf

f = vdf.parse(open('npc_heroes.txt'))

print vdf.dump(f['DOTAHeroes']['npc_dota_hero_zuus']['Bot'],pretty=True)
```

You can see that the output, specifically for Build/Loadout, is out of order vs the .txt file and duplicated items (e.g... Blades of Attack, used for Phase Boots) are omitted. This likely affects Javascript in a similar way, though I believe PHP handles this a bit better and should be fine. 

Let me know if this is something you're working on or if I should take a crack at it and submit a pull. 

Thanks!

**Comment by rossengeorgiev** [comment]
Interesting. I wasn't aware of the use of duplicates keys in VDF files. The order can easily be sorted out by using `OrderedDict` instead, but the duplicates pose a challenge. 

One way to would be to have the values of duplicates be a `list`. That would ruin the order if they are not one after another. However, the biggest drawback would be using the structure. You have to explicitly handle whenever a value is string or `list`. That seems like a messy approach. Perhaps it's better to detect collision when parsing the file, and append an index number resulting in a dictionary like this:

```python
{
"item_tango": "ITEM_CONSUMABLE | ITEM_SELLABLE"
"item_tango1": "ITEM_CONSUMABLE | ITEM_SELLABLE"
"item_tango2": "ITEM_CONSUMABLE | ITEM_SELLABLE"
}
```
This seems like the most elegant option. Since duplicate keys are not common, they can be a special case, which leaves it to the application to handle.

**Comment by tzodyaq** [comment]
I was comparing the Python version to the PHP version, which appears to function as expected. PHP var_dump shows that it uses arrays of arrays, which both preserves order and allows for duplicate entries. Would something like that work for Python?

**Comment by rossengeorgiev** [comment]
If we want to fix it for all languages we need to make a hack. In my opinion, appending and index to the colliding keys, is the best approach. It will work on all languages. Since this is not common case in VDF file, if we use that approach, we could make function that when given a key will look for the duplicates and return a list.

**Comment by rossengeorgiev** [comment]
These issues have been addressed in `v1.4` of the python `vdf` module, by being able to set a custom mapper class. This doesn't solve the issue with duplicate keys, but allows for using custom mapper that can handle that particular case.

**Comment by rossengeorgiev** [comment]
Here is an example implementation in Python for handling duplicate keys and preserving order:

https://github.com/rossengeorgiev/dota2_notebooks/blob/master/DuplicateOrderedDict_for_VDF.ipynb
